lib/src/main/java/io/ably/lib/types/ClientOptions.java

```diff
@@ -194,4 +195,6 @@ public ClientOptions(String key) throws AblyException {
 	 * before responding.
 	 */
 	public boolean pushFullWait = false;
+
+	public final Map<String, PluggableCodec> Codecs = new HashMap<>();
```

Comment:
The way it is now client code using "new" codecs would work against older versions of the libraries that do not support these, too. As mentioned above they will just decode what is possible to decode and return the intermediate result. Customer code can then decode a step and return the rest to the lib for further decoding. Having an enum we lose that workflow.

Reply:
I'm not sure that's true, or at least not if we go with this being a 'plugin' rather than a 'codec'. Where, for a plugin, there's assumed to be more internal wiring required in a new release to make it work therefore it can't simply be supplied to an older library.

Reply:
Currently it is a codec. If we want to make it a plugin then a lot of things will get very different. As mentioned above not really sure generic plugins should be introduced as part of this effort to replace the generic codecs that were also part of this effort and we gave up on.

Reply:
Can you expand on what needs to be very different? The codec / plugin interface is empty so I thought this was just a matter of naming.